Allow alter type ALTER_UPGRADE to call SolvAddExcludes()

sidcha opened the pull request:
`TDNFPrepareAllPackages()` forces `nAlterType` to `ALTER_UPGRADE` for both `ALTER_UPGRADE` and `ALTER_UPGRADEALL` when `--security` is passed. Subsequently, when `TDNFGoal()` is called, the excludes are skipped from the job queue in `TDNFAddGoal()` but are not marked in solver map as excludes with a call to `SolvAddExcludes()` as it's gating condition only check for `nAlterType == ALTER_UPGRADEALL`.

Fix this by allowing `nAlterType == ALTER_UPGRADE` to also call `SolvAddExcludes()`.

sidcha committed:
Remove unused Queue object

Signed-off-by: Siddharth Chandrasekaran <csiddharth@vmware.com>

sidcha committed:
Allow alter type ALTER_UPGRADE to call SolvAddExcludes()

TDNFPrepareAllPackages() forces nAlterType to ALTER_UPGRADE for both
ALTER_UPGRADE and ALTER_UPGRADEALL when --security is passed.
Subsequently, when TDNFGoal() is called, the excludes are skipped from the
job queue in TDNFAddGoal() but are not marked in solver map as excludes
with a call to SolvAddExcludes() as it's gating condition only check for
nAlterType == ALTER_UPGRADEALL.

Fix this by allowing nAlterType == ALTER_UPGRADE to also call
SolvAddExcludes().

Signed-off-by: Siddharth Chandrasekaran <csiddharth@vmware.com>

sidcha commented:
@keerthanakalyan, I want these commits merged as 2 separate commits for ease of blaming at a later time.

CC: @sshedi, @tapakund, @jaankit

ppadmavilasom reviewed (approved):
LGTM

keerthanakalyan commented on client/goal.c:
```diff
@@ -356,7 +356,8 @@ TDNFGoal(
 
-    if (nAlterType == ALTER_UPGRADEALL)
+    if (nAlterType == ALTER_UPGRADEALL ||
+        nAlterType == ALTER_UPGRADE)
```
This change fails for the packages which includes subpackages. For example if we provide --exclude=curl we get an error like below, since it is not able to exlcude curl-libs from the list.
```
]# ./tdnf update --security --exclude=curl
1. package curl-7.61.1-2.ph3.x86_64 is disabled
Found 1 problem(s) while resolving
Error(1301) : Solv general runtime error
```

sidcha replied:
That is was a valid error. We must skip this error where there are excludes.

sidcha committed:
Remove unnecessary condition checks in SkipBasedOnType()

Remove unnecessary condition checks in SkipBasedOnType() by leverage bit
mask's implicit AND-ing behavior.

Signed-off-by: Siddharth Chandrasekaran <csiddharth@vmware.com>

sshedi commented on solv/tdnfpackage.c:
```diff
@@ -1534,38 +1520,45 @@ SolvSplitEvr(
-        return (type == SOLVER_RULE_PKG_OBSOLETES ||
-                type == SOLVER_RULE_PKG_IMPLICIT_OBSOLETES ||
-                type == SOLVER_RULE_PKG_INSTALLED_OBSOLETES);
+        result = result || type == SOLVER_RULE_PKG_OBSOLETES ||
```
Cool 👍

sshedi reviewed (approved):
LGTM
